File: src/transformers/features/http/HttpValidateFormDataTransformer.ts
import { HttpValidateFormDataProgrammer } from "../../../programmers/http/HttpValidateFormDataProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpValidateFormDataTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.validateFormData",
      write: HttpValidateFormDataProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/HttpValidateFormDataTransformer.go
package http

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type httpValidateFormDataTransformerNamespace struct{}

var HttpValidateFormDataTransformer = httpValidateFormDataTransformerNamespace{}

func (httpValidateFormDataTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "http.validateFormData",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativehttpprogrammers.HttpValidateFormDataProgrammer.Write(x)
		},
	})
}
